clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2894,6 +2912,22 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::update_next_record(output_ordinal
         // sched_type_t::MAP_AS_PREVIOUSLY).
         return;
     }
+    // We modify the tid and pid fields to ensure uniqueness for
+    // core-sharded-on-disk and with analyzers not using our workload identifiers.
```

Comment:
> with analyzers not using our workload identifiers

I don't understand this. Analyzers can also use memtrace_stream_t::get_workload_id to get the workload identifiers right? (atleast in non-csod, which is what this part after "core-sharded-on-disk and ..." addresses I think)